src/main/java/ch/njol/skript/bukkitutil/ItemUtils.java

```diff
@@ -361,4 +358,36 @@ public static boolean isGlass(Material material) {
 				return false;
 		}
 	}
+
+	/**
+	 * Changes the ItemMeta of the provided ItemStack with the provided Consumer
+	 * @param itemStack ItemStack's ItemMeta to be changed
+	 * @param metaChanger Consumer to change the ItemMeta
+	 * @return ItemStack with the updated ItemMeta
+	 * @param <T>
+	 */
+	public static <T extends ItemMeta> ItemStack changeItemMeta(ItemStack itemStack, Consumer<T> metaChanger) {
+		//noinspection unchecked
+		T itemMeta = (T) itemStack.getItemMeta();
+		metaChanger.accept(itemMeta);
+		itemStack.setItemMeta(itemMeta);
+		return itemStack;
+	}
+
+	/**
+	 * Set's the item back from its original location, see {@link ItemUtils#asItemStack(Object)}
+	 * @param object The original object used to determine it's origin
+	 * @param itemStack The ItemStack to update the original location
+	 */
+	public static void setItem(Object object, ItemStack itemStack) {
+		ItemMeta itemMeta = itemStack.getItemMeta();
+		if (object instanceof Slot slot) {
+			slot.setItem(itemStack);
+		} else if (object instanceof ItemType itemType) {
+			itemType.setItemMeta(itemMeta);
+		} else if (object instanceof ItemStack itemStack1) {
+			itemStack1.setItemMeta(itemMeta);
+		}
```

Comment:
this should error if no valid type is matched

Reply:
What, like a Skript.error? or assert? or throw new IllegalState?

Reply:
probably an IAE